examples/topics/categorical/correlograms.py

```diff
@@ -0,0 +1,79 @@
+''' A categorical plot showing the correlations in mineral content for 214 samples of glass fragments
+obtained during forensic work.
+
+The dataset contains seven variables measuring the amounts of magnesium (Mg),
+calcium (Ca), iron (Fe), potassium (K), sodium (Na), aluminum (Al), and barium (Ba) found in each glass fragment.
+The colored circles represent the correlations between pairs of these variables.
+The magnitude of each correlation is encoded in the size of the circles.
+
+.. bokeh-example-metadata::
+    :sampledata: forensic_glass
+    :apis: bokeh.plotting.figure.scatter
+    :refs: :ref:`ug_topics_categorical_scatters`
+    :keywords: scatter, correlogram
+
+'''
+
+#from itertools import combinations
+import numpy as np
+import pandas as pd
+
+from bokeh.models import ColorBar, ColumnDataSource, FixedTicker, LinearColorMapper
+from bokeh.plotting import figure, show
+from bokeh.sampledata.forensic_glass import data as df
+from bokeh.transform import transform
+
+elements = ("Mg", "Ca", "Fe", "K", "Na", "Al", "Ba")
+pairs = []
+for i in range(len(elements)):
+    for j in range(len(elements) - 1, i, -1):
+        pair = (elements[i], elements[j])
+        pairs.append(pair)
+
+#pairs = list(combinations(elements, 2))
+
+x, y = list(zip(*pairs))
+
+correlations = []
+for x, y in pairs:
+    matrix = np.corrcoef(df[x], df[y])
+    correlations.append(matrix[0, 1])
+
+new_df = pd.DataFrame({
+    "oxide_1": [x[0] for x in pairs],
+    "oxide_2": [x[1] for x in pairs],
+    "correlation": correlations,
+    "dot_size": [abs(corr) * 120 for corr in correlations],
```

Comment:
Currently the small circles seem very small, is it possible to increase this without making the big circles too big? Do we know what kind of scale transform the book used?

Reply:
This was an issue for me when I was creating this plot for the first time. The actual correlation values are almost zero for the small circles, so getting them to actually show would require a much larger expansion factor that would make the larger circles too large.

I don't know the scale the book used but i suspect it may not be a uniform scale. Maybe different scaling for different size ranges?

Reply:
I suspected that might be the case. I think it wouldn't hurt to try some simple alternative scalings. Offhand, maybe a log of the percentage? e.g.`log(100*x + 1)` (the +1 just to keep things nice/positive)

Reply:
I found a scaling factor I think works well now: `(1+ 10 * abs(x)) * 10` which resulted in a plot like this with the circle borders in black

<img width="611" alt="Screenshot 2023-10-05 at 8 33 07 AM" src="https://github.com/bokeh/bokeh/assets/99359668/02ee350c-fcd6-4466-8bc7-bee28a5f06bb">

Reply:
I think this looks good now, the border makes it much easier to read! Other tiny tweaks hat might be worth a quick look:

* alot of our plots have `#fefefe` or `#fafafa` backgrounds, how does this plot look with a light grey background?
* can the border be softened slightly and still provide the benefit? e.g. maybe a dark gray, or slight alpha, or even seeing what a dashed or dotted line looks like